### DISCUSSION TITLE
Running a step only if the deploy resulted in a new commit (changed files)

### DISCUSSION THREAD

**Discussion by tiagoad** [JamesIves/github-pages-deploy-action#494] [Q&A] [answered]
Hi,

I want to run a few steps, only if the deploy resulted in changes. Is there any way to do this?

Thanks!

**Comment by JamesIves** [comment]
The action emits deployment in the form of an environment variable that you can use to conditionally move your workflow onto the next step: https://github.com/JamesIves/github-pages-deploy-action#deployment-status

**Comment by emmahsax** [comment, marked as answer]
Try something like this:

```yml
    - name: Deploy Site
      uses: JamesIves/github-pages-deploy-action@releases/v3
      with:
        .
        .
        .

    - name: Continue
      if: ${{ env.DEPLOYMENT_STATUS }} == "success"
      run: echo "Continuing because deploy happened"

**Reply by Pike** [reply]
v4 is also going to have this result as a step output, which you can also use. If there are a couple of steps, you might be able to pull them into a separate job, and run the job only if the previous one succeeded (aka, not fail nor skipped). That's not on the releases/v4 branch yet, though.
